src/api/server.rkt

```diff
@@ -120,29 +120,71 @@
 
 (define (manager-ask msg . args)
   (log "Asking manager: ~a, ~a.\n" msg args)
-  (match msg
-    ['wait (manager-ask-with-callback msg args)]
-    ['result (manager-ask-with-callback msg args)]
-    ['timeline (manager-ask-with-callback msg args)]
-    ['check (manager-ask-with-callback msg args)]
-    ['count (manager-ask-with-callback msg args)]
-    ['improve (manager-ask-with-callback msg args)]
-    [_ (place-channel-put manager (flatten (list msg args)))]))
+  (if manager
+      (match msg
+        ['wait (manager-ask-with-callback msg args)]
+        ['result (manager-ask-with-callback msg args)]
+        ['timeline (manager-ask-with-callback msg args)]
+        ['check (manager-ask-with-callback msg args)]
+        ['count (manager-ask-with-callback msg args)]
+        ['improve (manager-ask-with-callback msg args)]
+        [_ (place-channel-put manager (flatten (list msg args)))])
+      (match msg ; public commands
```

Comment:
You can do something like `(match (list* msg args)` and then the cases will be things like `[(list 'wait hash-false job-id)` and then you don't need an extra `match-define` line in each one.